mountpoint-s3-client/src/object_client.rs

```diff
@@ -266,6 +266,18 @@ pub enum GetObjectAttributesError {
     NoSuchKey,
 }
 
+/// Server side encryption configuration for newly created objects
+#[derive(Debug, Default, Clone)]
+pub enum ServerSideEncryption {
```

Comment:
I generally like the idea of keeping things alike the SDK, but our way here, with enum, offers a better developer experience isn't it? 

Passing a enum makes it impossible to set a key ID without choosing a proper encryption type. We can not avoid adding an additional type here as well, as the SDK itself has a enum for SSE types, it just does not contain a key. I'd say observing (with the help of IDE tooling) that you may need to set a key `ServerSideEncryption::Kms{ key_id: None }` instead of `ServerSideEncryption::Kms` is a low price for the described benefit.

Reply:
I don't have a super strong opinion on the enum, I do see your point with the benefits. Maybe you have a preference, @passaro?

Reply:
The Rust SDK also uses 2 properties: https://docs.rs/aws-sdk-s3/latest/aws_sdk_s3/operation/put_object/struct.PutObjectInput.html#structfield.server_side_encryption

I lean slightly for consistency in this case.

Reply:
Fixed to comply with rust SDK, `ServerSideEncryption` may be later moved out from the client to `mountpoint-s3`